microprofile/grpc/core/src/main/java/io/helidon/microprofile/grpc/core/Instance.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2019, 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.grpc.core;
+
+import java.util.Objects;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+import io.grpc.Status;
+
+/**
+ * A supplier of instances of objects.
+ */
+public interface Instance {
```

Comment:
This is a very fishy thing. It uses reflection, name conflicts with CDI, and at least part of it is implemented in `LazyValue` in Helidon.
I would again question if this is needed, and if it is the right way to achieve what we want.

Reply:
Renamed for now